src/Platforms/SQLitePlatform.php

```diff
@@ -905,7 +921,7 @@ private function getForeignKeysInAlteredTable(TableDiff $diff): array
                     ->getValue(),
             );
 
-            unset($foreignKeys[$constraintKey]);
+            unset($foreignKeys[$keysByName[$constraintKey]]);
```

Comment:
`$keysByName[$constraintKey]` isn't guaranteed to be set (dropping a FK that doesn't exist). This should  be an exception. Let's make an it assertion for now and revisit later.